### ISSUE TITLE
Deal with Vega-Lite specific mark config

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#1995]
With the current implementation of Vega Mark Config, any config value inside a Vega mark config or role config will pollute the scene graph.

**Comment by kanitw** [comment]
A few solutions:

1) Redesign Vega-Lite config to have these config properties outside. This was the solution that we discussed in the meeting, but upon closer inspection -- I start to think this solution is not good because it will force us to design weird config options.   

For example, where should we move `config.mark.color` to?  `config.markColor` or `config.vl.markColor` would be confusing for users why should color be different from other mark properties.  

2) Make Vega-Lite remove all of these properties. This approach require us to make VL API output a new config object that users need to use as input to the vega-parser rather than the original config.  

However, I don't like this for two reasons:
- This require many custom checks for different types of configs (which is gonna be way more code than in solution 3.) 
- More importantly, so far we avoid modifying config object as it will make the Vega-Lite API pretty complicated. 

3) Make vega-parser's encode-util [skip](https://github.com/vega/vega-parser/blob/master/src/parsers/encode/encode-util.js#L50)s non-mark properties. 

Basically,  we could add another line for the `skip` variable to check if the key is a member of the `encodeEntry` schema defined in the [schema generation code](https://github.com/vega/vega-parser/blob/master/schema/encode.js#L224).  

I can see that one drawback of this approach is that `src/` will depend on a part of `schema/` but besides that it's relatively simple to implement.  So if haven't overlooked anything and this works, 
it seems like the best option to me. 

cc: @domoritz @jheer

**Comment by kanitw** [comment]
Depends on https://github.com/vega/vega-parser/issues/23

**Comment by kanitw** [comment]
Looks like we cannot solve this in Vega-parser. 

We will need a utility function that takes a Vega-Lite config as input, and return a new config object that has no Vega-Lite specific mark config.

**Comment by kanitw** [comment]
Fixed in #2372
